internal/brokers/withexamples.go

```diff
@@ -0,0 +1,116 @@
+//go:build withexamplebroker
+
+package brokers
+
+import (
+	"context"
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"path/filepath"
+	"time"
+
+	"github.com/ubuntu/authd/internal/brokers/examplebroker"
+	"github.com/ubuntu/authd/internal/log"
+)
+
+// useExampleBrokers starts a mock system bus and exports the examplebroker in it.
+func useExampleBrokers() (string, error) {
+	busCleanup, err := startSystemBusMock()
+	if err != nil {
+		return "", err
+	}
+	log.Debugf(context.Background(), "Mock system bus started on %s\n", os.Getenv("DBUS_SYSTEM_BUS_ADDRESS"))
+
+	// Create the directory for the broker configuration files.
+	cfgPath, err := os.MkdirTemp(os.TempDir(), "examplebroker.d")
+	if err != nil {
+		busCleanup()
+		return "", err
+	}
+	cleanup := func() {
+		os.RemoveAll(cfgPath)
+		busCleanup()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	done := make(chan struct{})
+	go func() {
+		defer close(done)
+		if err = examplebroker.StartBus(ctx, cfgPath); err != nil {
+			fmt.Printf("Error starting example broker: %v\n", err)
+			cleanup()
```

Comment:
as you are doig the cleanup, calling buscleanup in some cases here too, I would just defer the call to busCleanup and os.RemoveAll(cfgPath) after the defer close(done) (to be executed before). That way, we don’t have to wonder what’s happening between busCleanup and cleanup and it’s in a single place for cleaning up the bus.